actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/BatchOperationNotifierActor.java

```diff
@@ -0,0 +1,335 @@
+ *
+ * @author github.com/iostream04
+ */
+
+/** @author mayank */
+@ActorConfig(
+  tasks = {"batchBulk", "batchUpdate", "batchOperation"},
+  asyncTasks = {"batchBulk", "batchUpdate", "batchOperation"}
+)
+public class BatchOperationNotifierActor extends BaseActor {
+  private static CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+
+  private DecryptionService decryptionService =
+      org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.getDecryptionServiceInstance(
+          null);
+
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    if (request.getOperation().equalsIgnoreCase(ActorOperations.BATCH_BULK.getValue())) {
+      Map<String, Object> requestMap = request.getRequest();
+      batchCreateOperationNotifier(
+          (CourseBatch) requestMap.get(BackgroundOperations.COURSE_BATCH.name()),
+          (String) requestMap.get(BackgroundOperations.OPERATION_TYPE.name()));
+    } else if (request.getOperation().equalsIgnoreCase(ActorOperations.BATCH_UPDATE.getValue())) {
+      Map<String, Object> requestMap = request.getRequest();
+      batchUpdateOperationNotifier(
+          (CourseBatch) requestMap.get(BackgroundOperations.OLD.name()),
+          (CourseBatch) requestMap.get(BackgroundOperations.NEW.name()));
+    } else if (request
+        .getOperation()
+        .equalsIgnoreCase(ActorOperations.BATCH_OPERATION.getValue())) {
+      Map<String, Object> requestMap = request.getRequest();
+      batchEnrollOperationNotifier(
+          (Map<String, Object>) requestMap.get(BackgroundOperations.COURSE_MAP.name()),
+          (CourseBatch) requestMap.get(BackgroundOperations.COURSE_BATCH.name()),
+          (String) requestMap.get(BackgroundOperations.OPERATION_TYPE.name()));
+    } else {
+      onReceiveUnsupportedOperation(request.getOperation());
+    }
+  }
+
+  /*
+   * This method takes courseBatch and operationType String as parameter and
+   * process invite-only batch enrol and unenroll operation notifications mentor
+   * and learner add remove Operation
+   */
+  private void batchCreateOperationNotifier(CourseBatch courseBatch, String operationType) {
+    if (operationType.equals(JsonKey.ADD)) {
+      sendEmailNotification(courseBatch, JsonKey.ADD);
+    } else if (operationType.equals(JsonKey.REMOVE)) {
+      sendEmailNotification(courseBatch, JsonKey.REMOVE);
+    }
+  }
+
+  /*
+   * This method takes map and courseBatch as parameter and process open batch
+   * enrol and unenroll operation notifications
+   */
+  @SuppressWarnings("unchecked")
+  public void batchEnrollOperationNotifier(
+      Map<String, Object> courseMap, CourseBatch courseBatch, String operationType) {
+    Map<String, Object> requestMap = createRequestMap(courseBatch);
+    List<String> userId = new ArrayList<>();
+    userId.add((String) courseMap.get(JsonKey.USER_ID));
+    List<Map<String, Object>> participentList = getUsersFromDB(userId);
```

Comment:
Email will come as Encrypted format , need to decrypt it.

Reply:
actually we are doing this and we have tested it also